lib/autocomplete.ts

```diff
@@ -0,0 +1,93 @@
+import {dom, DomElementArg, onElem, onKeyElem} from 'grainjs';
+import {MaybeObsArray} from 'grainjs';
+import {BaseMenu, defaultMenuOptions, IMenuOptions, menuItem} from './menu';
+import {IOpenController, IPopupContent, PopupControl, setPopupToFunc} from './popup';
+
+/**
+ * User interface for creating a weasel autocomplete element in DOM.
+ *
+ * Usage:
+ *      const employees = ['Thomas', 'June', 'Bethany', 'Mark', 'Marjorey', 'Zachary'];
+ *      const inputElem = input(...);
+ *      autocomplete(inputElem, employees);
+ */
+export function autocomplete<T>(
+  inputElem: HTMLInputElement,
+  optionArray: MaybeObsArray<string>,
+  options: IMenuOptions = {}
+): HTMLInputElement {
+
+  // Options to pass into the Autocomplete class.
+  const menuOptions: IMenuOptions = {
+    ...defaultMenuOptions,
+    menuCssClass: options.menuCssClass,
+    trigger: [(triggerElem: Element, ctl: PopupControl) => {
+      dom.onElem(triggerElem, 'click', () => ctl.open())
+    }],
+    stretchToSelector: 'input'
+  };
+
+  // DOM content of the open autocomplete menu.
+  const menuContent = () => [
+    dom.forEach(optionArray, (option) =>
+      menuItem(() => { inputElem.value = option; },
+        option
+      )
+    )
+  ];
+
+  setPopupToFunc(inputElem,
+    (ctl) => Autocomplete.create(null, ctl, menuContent(), menuOptions),
+    menuOptions);
+
+  return inputElem;
+}
+
+/**
+ * Creates an instance of Menu meant to be attached to an input and used as an autocomplete.
+ *
+ * Should always be created using autocomplete(), which accepts as an argument the input element.
```

Comment:
If it should always be created using `autocomplete()`, then maybe `Autocomplete` shouldn't be exported?

Reply:
Yes